source/common/router/route_config_update_receiver_impl.cc

```diff
@@ -42,36 +40,42 @@ bool RouteConfigUpdateReceiverImpl::onVhdsUpdate(
     const Protobuf::RepeatedPtrField<std::string>& removed_resources,
     const std::string& version_info) {
 
-  std::map<std::string, envoy::config::route::v3::VirtualHost> vhosts_before_this_update(
-      vhds_virtual_hosts_);
-  const bool removed = removeVhosts(vhds_virtual_hosts_, removed_resources);
-  const bool updated = updateVhosts(vhds_virtual_hosts_, added_vhosts);
+  auto vhosts_before_this_update =
+      std::make_unique<std::map<std::string, envoy::config::route::v3::VirtualHost>>(
+          *vhds_virtual_hosts_);
+  const bool removed = removeVhosts(*vhds_virtual_hosts_, removed_resources);
+  const bool updated = updateVhosts(*vhds_virtual_hosts_, added_vhosts);
 
-  envoy::config::route::v3::RouteConfiguration route_config_before_this_update;
-  route_config_before_this_update.CopyFrom(route_config_proto_);
-  rebuildRouteConfig(rds_virtual_hosts_, vhds_virtual_hosts_, route_config_proto_);
+  auto route_config_before_this_update =
+      std::make_unique<envoy::config::route::v3::RouteConfiguration>();
+  route_config_before_this_update->CopyFrom(*route_config_proto_);
+  rebuildRouteConfig(rds_virtual_hosts_, *vhds_virtual_hosts_, *route_config_proto_);
```

Comment:
Could rebuildRouteConfig itself throw?  Do we need to also undo changes to vhds_virtual_hosts_ and route_config_proto_ if it does?

Reply:
> Could rebuildRouteConfig itself throw

Don't think so.